aquadoggo/src/network/service.rs

```diff
@@ -140,13 +110,34 @@ pub async fn network_service(
                 } => {
                     debug!("ConnectionClosed: {peer_id} {endpoint:?} {num_established} {cause:?}")
                 }
+                SwarmEvent::ConnectionEstablished { peer_id, .. }
+                    if network_config.rendezvous_client
+                        && network_config.rendezvous_peer_id.unwrap() == peer_id =>
```

Comment:
We should always be safe to unwrap the `rendezvous_peer_id` here because the CLI will ensure it's provided if `rendezvous_client` was set to `true`.